### ISSUE TITLE
GraphErrorChecker#response_hash returns nil causing gem to crash

### ISSUE THREAD

**Issue by nasaabg** [arsduo/koala#673]
This is a Koala bug/issue problem.

Hi!
At @zendesk we use your gem in version `3.0.0` and I come across that issue on our Production ENV.
For some reason we have been notified about an error:

```
NoMethodError
undefined method `[]' for nil:NilClass
```

in `koala/api/graph_error_checker.rb in base_error_info at line 57`

<img width="1056" alt="Screenshot 2022-07-21 at 17 40 45" src="https://user-images.githubusercontent.com/2679054/180320049-02170053-434b-49f1-b22e-aa352442923e.png">

I did a little of debugging and seems that https://github.com/arsduo/koala/blob/master/lib/koala/api/graph_error_checker.rb#L60 returns `nil` in some cases while I assume we expect `#response_hash` always return Hash (empty or not).

Cases like those would return sth different than Hash and would cause gem to crash:
* JSON.parse('true') ==> true
* JSON.parse('null') ==> `nil` (our case)
* JSON.parse('false') ==> false

I have a proposal to fix it, but I can not `push` my branch to the repository and open PR :( 
So here it is:
```ruby
def response_hash
  # Normally, we start with the response body. If it isn't valid JSON, we start with an empty
  # hash and fill it with error data.
  @response_hash ||= begin
    response_body = JSON.parse(body)
    response_body.is_a?(Hash) ? response_body : {}
  rescue JSON::ParserError
    {}
  end
end
```

**NOTE**: Looks like we are using `fork` but this issues is still valid

**Comment by nasaabg** [comment]
@ylecuyer any chance you may look into it? We would love to start using the original gem at Zendesk

**Comment by ylecuyer** [comment]
I missed that one, I have opened https://github.com/arsduo/koala/pull/681 and will make a release in the coming days (depending on https://github.com/arsduo/koala/pull/680)

For next time, if you want to be able to create a PR on the project you can fork it, make the change on your fork, then github will propose to open a PR.

**Comment by ylecuyer** [comment]
released as 3.5.0
